examples/screen_cast_gstreamer.rs

```diff
@@ -0,0 +1,80 @@
+use ashpd::desktop::screencast::{CursorMode, Screencast, SourceType, Stream};
+use gst::prelude::*;
+
+use std::os::fd::{AsRawFd, OwnedFd};
+
+async fn open_portal() -> ashpd::Result<(Stream, OwnedFd)> {
+    let proxy = Screencast::new().await?;
+    let session = proxy.create_session().await?;
+    proxy
+        .select_sources(
+            &session,
+            CursorMode::Embedded,
+            SourceType::Monitor | SourceType::Window | SourceType::Virtual,
+            false,
+            None,
+            ashpd::desktop::PersistMode::ExplicitlyRevoked,
+        )
+        .await?;
+
+    let response = proxy.start(&session, None).await?.response()?;
+    let stream = response
+        .streams()
+        .first()
+        .expect("No stream found or selected")
+        .to_owned();
+
+    let fd = proxy.open_pipe_wire_remote(&session).await?;
+
+    Ok((stream, fd))
+}
+
+#[tokio::main]
+async fn main() {
+    gst::init().unwrap();
+
+    let (stream, stream_fd) = open_portal().await.unwrap();
+    let pipewire_node_id = &stream.pipe_wire_node_id();
+    let stream_raw_fd = &stream_fd.as_raw_fd();
+
+    let pipewire_element = gst::ElementFactory::make("pipewiresrc")
```

Comment:
An alternative approach to get a pipewiresrc with the correct node and fd set is to construct a pipewiredeviceprovider with the fd and let it give you the pipewiresrc. See [1, 2] for example. Up to you whether this makes sense for the example.

[1] https://gitlab.gnome.org/GNOME/snapshot/-/blob/main/aperture/src/device_provider.rs?ref_type=heads#L109
[2] https://gitlab.gnome.org/GNOME/snapshot/-/blob/main/aperture/src/device_provider.rs?ref_type=heads#L119